### PR TITLE
[BPK-3838]: Add icon to chip

### PR DESCRIPTION
**Pull request by olliecurtis**
# Background

This PR adds `leadingAccessoryView` prop to `BpkChip` to support adding icons at the begining of the component.

# Changes
- Added new prop `leadingAccessoryView` to accept `BpkIcon` of consumer choice
- Added default story and dismissible story with leading icon 
- Added tests, screenshots LM and DM
- Updated docs with prop

### CODE REVIEW

File: lib/bpk-component-chip/README.md [hunk 1]
```diff
@@ -78,10 +78,15 @@ export default class App extends Component {
 | selected              | bool                                                                      | false    | false         |
 | innerChipStyle        | ViewStyle                                                                 | false    | null          |
 | type                  | oneOf(CHIP_TYPES)                                                         | false    | primary       |
+| leadingAccessoryView  | element                                                                   | false    | null          |
 
 **Note that on Android, `style` should be used for positional styling and `innerChipStyle` should be used for other styling.**
 (On iOS, `style` and `innerChipStyle` are applied together so it doesn't matter which you use!)
 
+## `accessoryView`
```
Comment (shaundon) [inline lib/bpk-component-chip/README.md:86]:
```suggestion
## `leadingAccessoryView`
```

File: lib/bpk-component-chip/README.md [hunk 2]
```diff
@@ -99,6 +104,11 @@ export default class App extends Component {
 | disabled              | bool                                                                      | false    | false         |
 | innerChipStyle        | ViewStyle                                                                 | false    | null          |
 | type                  | oneOf(CHIP_TYPES)                                                         | false    | primary       |
+| leadingAccessoryView  | element                                                                   | false    | null          |
 
 **Note that on Android, `style` should be used for positional styling and `innerChipStyle` should be used for other styling.**
 (On iOS, `style` and `innerChipStyle` are applied together so it doesn't matter which you use!)
+
+## `accessoryView`
```
Comment (shaundon) [inline lib/bpk-component-chip/README.md:112]:
```suggestion
## `leadingAccessoryView`
```

File: lib/bpk-component-chip/src/BpkChipWrapper.js [hunk 3]
```diff
@@ -121,11 +121,15 @@ const dynamicStyles = BpkDynamicStyleSheet.create({
...
+  textWithLeadingAccessoryView: {
+    marginStart: spacingSm,
+  },
 });
 
 type Props = {
   ...$Exact<CommonProps>,
   dismissible: boolean,
+  leadingAccessoryView: ?Element<*>,
   selected: boolean,
```
Comment (shaundon) [inline lib/bpk-component-chip/src/BpkChipWrapper.js:133]: Nitpick: Our convention is to have optional types below the non-optional ones.

```suggestion
  selected: boolean,
  leadingAccessoryView: ?Element<*>,
```

**Comment by shaundon** [comment]
Couple of small things but otherwise looking great, as far as I can see 💯

File: lib/bpk-component-chip/README.md (hunk 1 above)
Comment (olliecurtis) [reply lib/bpk-component-chip/README.md]: Addressed!

File: lib/bpk-component-chip/README.md (hunk 2 above)
Comment (olliecurtis) [reply lib/bpk-component-chip/README.md]: Addressed!

File: lib/bpk-component-chip/src/BpkChipWrapper.js (hunk 3 above)
Comment (olliecurtis) [reply lib/bpk-component-chip/src/BpkChipWrapper.js]: Addressed!

**Pull request merged**